sfs/mono/source.py

```diff
@@ -705,6 +705,101 @@ def plane_averaged_intensity(omega, x0, n0, grid, c=None):
     return util.XyzComponents([i * n for n in n0])
 
 
+def pulsating_sphere_displacement(omega, x0, a, d, grid, c=None):
+    """Particle displacement of a pulsating sphere.
+
+    Paramters
+    ---------
+    omega : float
+        Frequency of pulsating sphere.
+    x0 : (3,) array_like
+        Position of source.
+    a : float
+        Radius of sphere.
+    d : float
+        Amplitude of displacement.
+    grid : triple of array_like
+        The grid that is used for the sound field calculations.
+        See `sfs.util.xyz_grid()`.
+    c : float, optional
+        Speed of sound.
+
+    Returns
+    -------
+    `XyzComponents`
+        Particle displacement at positions given by *grid*.
+    """
+    k = util.wavenumber(omega, c)
+    x0 = util.asarray_1d(x0)
+    offset = grid - x0
+    r = np.linalg.norm(offset)
+    r[r <= a] = np.nan
+    return util.XyzComponents([d * a / r**2 * np.exp(-1j * k * (r - a)) * o
+                               for o in offset])
```

Comment:
`XyzComponents` has an `.apply()` method that you could probably use here: http://python.sfstoolbox.org/en/latest/utilities.html#sfs.util.XyzComponents.apply

Reply:
That's a very useful feature.
But how can I use it in this case? Should I define a new function? Or is there a more pythonic way?

Reply:
I'm not sure if it's really worth to use `.apply()` in this case ...

But if you do, you could use a `lambda` function instead of defining a separate function before.
However, it is uncommon to have a line break in a `lambda` function. Line breaks in list comprehensions are more common.

Probably you should just leave it as it is ...?

Reply:
I would keep it as it is, but please let me know if you come up with a better idea.

Reply:
OK.